clients/destination.go

```diff
@@ -124,12 +125,18 @@ func (c *DestinationClient) newManagedClient(ctx context.Context, path string) (
 	c.wg.Add(1)
 	go func() {
 		defer c.wg.Done()
-		scanner := bufio.NewScanner(reader)
-		for scanner.Scan() {
+		lr := newLogReader(reader)
+		for {
+			line, err := lr.NextLine()
+			if errors.Is(err, io.EOF) {
+				break
+			} else if err != nil {
```

Comment:
can we somehow distinguish between line-to-long to some other errors where we actually want to exit.